Add pre-commit hooks for development workflow

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

We should add Pre-commit hooks to clean up and standardize the code using black and flake8.

### Describe the solution you'd like.

_No response_

### Describe an alternate solution.

_No response_

### Anything else? (Additional Context)

_No response_

These are all the problems flake8 is highlighting

```

mindsdb/integrations/handlers/file_handler/file_handler.py:141:5: C901 'FileHandler._get_data_io' is too complex (19)
mindsdb/integrations/handlers/ludwig_handler/ludwig_handler.py:10:1: F401 'mindsdb_sql.parser.ast.BinaryOperation' imported but unused
mindsdb/integrations/handlers/ludwig_handler/ludwig_handler.py:10:1: F401 'mindsdb_sql.parser.ast.Constant' imported but unused
mindsdb/integrations/handlers/ludwig_handler/ludwig_handler.py:10:1: F401 'mindsdb_sql.parser.ast.Show' imported but unused
mindsdb/integrations/handlers/ludwig_handler/ludwig_handler.py:88:13: F401 'ludwig as lw' imported but unused
mindsdb/integrations/handlers/ludwig_handler/ludwig_handler.py:89:13: F401 'ludwig.api.LudwigModel' imported but unused
mindsdb/integrations/handlers/ludwig_handler/ludwig_handler.py:90:13: F401 'ludwig.automl.auto_train' imported but unused
mindsdb/migrations/versions/2022-07-15_b5b53e0ea7f8_training_data_rows_columns_count.py:11:1: F401 'mindsdb.interfaces.storage.db' imported but unused
mindsdb/integrations/handlers/clickhouse_handler/clickhouse_handler.py:3:1: F401 'clickhouse_driver' imported but unused
mindsdb/api/mysql/mysql_proxy/data_types/mysql_packets/command_packet.py:15:1: F401 'mindsdb_sql.parse_sql' imported but unused
mindsdb/api/mysql/mysql_proxy/data_types/mysql_packets/command_packet.py:16:1: F401 'mindsdb_sql.parser.ast.Parameter' imported but unused
mindsdb/integrations/handlers/lightwood_handler/tests/test_lightwood_handler.py:14:1: E402 module level import not at top of file
mindsdb/integrations/handlers/lightwood_handler/tests/test_lightwood_handler.py:18:1: E402 module level import not at top of file
mindsdb/integrations/handlers/lightwood_handler/tests/test_lightwood_handler.py:19:1: E402 module level import not at top of file
mindsdb/integrations/handlers/lightwood_handler/tests/test_lightwood_handler.py:22:1: E402 module level import not at top of file
mindsdb/integrations/handlers/lightwood_handler/tests/test_lightwood_handler.py:23:1: F401 'mindsdb.integrations.utilities.test_utils.HandlerControllerMock' imported but unused
mindsdb/integrations/handlers/lightwood_handler/tests/test_lightwood_handler.py:23:1: E402 module level import not at top of file
mindsdb/integrations/handlers/lightwood_handler/tests/test_lightwood_handler.py:28:1: E402 module level import not at top of file
mindsdb/integrations/handlers/lightwood_handler/tests/test_lightwood_handler.py:29:1: E402 module level import not at top of file
mindsdb/integrations/handlers/lightwood_handler/tests/test_lightwood_handler.py:32:1: E402 module level import not at top of file
mindsdb/integrations/handlers/lightwood_handler/tests/test_lightwood_handler.py:33:1: E402 module level import not at top of file
mindsdb/integrations/handlers/lightwood_handler/tests/test_lightwood_handler.py:34:1: E402 module level import not at top of file
mindsdb/integrations/handlers/hive_handler/hive_handler.py:9:1: F401 'sqlalchemy.create_engine' imported but unused
mindsdb/migrations/versions/2022-08-25_6a54ba55872e_view_integration.py:11:1: F401 'mindsdb.interfaces.storage.db' imported but unused
mindsdb/integrations/utilities/utils.py:3:1: F401 'mindsdb_sql.parser.ast.Join' imported but unused
mindsdb/integrations/utilities/utils.py:3:1: F401 'mindsdb_sql.parser.ast.Select' imported but unused
mindsdb/integrations/utilities/utils.py:3:1: F401 'mindsdb_sql.parser.ast.Star' imported but unused
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_query.py:32:13: E713 test for membership should be 'not in'
mindsdb/api/http/namespaces/util.py:12:1: F401 'mindsdb.utilities.log.log' imported but unused
mindsdb/api/mongo/responders/buildinfo.py:1:1: F401 'mindsdb.__about__.__version__ as mindsdb_version' imported but unused
tests/prediction_latency_test/test.py:25:26: F821 undefined name 'Predictor'
tests/prediction_latency_test/test.py:39:26: F821 undefined name 'Predictor'
mindsdb/integrations/handlers/s3_handler/s3_handler.py:10:1: F401 'mindsdb_sql.render.sqlalchemy_render.SqlalchemyRender' imported but unused
mindsdb/integrations/handlers/s3_handler/s3_handler.py:129:21: F841 local variable 'stats' is assigned to but never used
mindsdb/integrations/handlers/mssql_handler/mssql_handler.py:1:1: F401 'contextlib.closing' imported but unused
mindsdb/integrations/handlers/datastax_handler/datastax_handler.py:2:1: F401 '..scylla_handler.connection_args' imported but unused
mindsdb/api/mongo/responders/find.py:2:1: F403 'from mindsdb_sql.parser.ast import *' used; unable to detect undefined names
mindsdb/api/mongo/responders/find.py:71:24: F405 'Join' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mysql/mysql_proxy/classes/com_operators.py:20:40: W605 invalid escape sequence '\s'
mindsdb/api/mysql/mysql_proxy/classes/com_operators.py:20:42: W605 invalid escape sequence '\S'
mindsdb/interfaces/database/integrations.py:3:1: F401 'os' imported but unused
mindsdb/api/mysql/mysql_proxy/libs/constants/mysql.py:20:1: E302 expected 2 blank lines, found 1
mindsdb/integrations/handlers/mlflow_handler/mlflow_handler/mlflow_handler.py:17:1: F401 'mindsdb.utilities.config.Config' imported but unused
mindsdb/integrations/handlers/mlflow_handler/mlflow_handler/mlflow_handler.py:228:12: E713 test for membership should be 'not in'
mindsdb/integrations/handlers/mlflow_handler/mlflow_handler/mlflow_handler.py:230:14: E713 test for membership should be 'not in'
mindsdb/integrations/handlers/lightwood_handler/lightwood_handler/lightwood_handler.py:17:1: F401 'mindsdb_sql.parser.ast.BinaryOperation' imported but unused
mindsdb/integrations/handlers/lightwood_handler/lightwood_handler/lightwood_handler.py:17:1: F401 'mindsdb_sql.parser.ast.Constant' imported but unused
mindsdb/integrations/handlers/lightwood_handler/lightwood_handler/lightwood_handler.py:424:5: C901 'LightwoodHandler.predict' is too complex (73)
mindsdb/__main__.py:13:1: E402 module level import not at top of file
mindsdb/__main__.py:15:1: E402 module level import not at top of file
mindsdb/__main__.py:16:1: E402 module level import not at top of file
mindsdb/__main__.py:17:1: E402 module level import not at top of file
mindsdb/__main__.py:18:1: E402 module level import not at top of file
mindsdb/__main__.py:19:1: E402 module level import not at top of file
mindsdb/__main__.py:20:1: E402 module level import not at top of file
mindsdb/__main__.py:21:1: E402 module level import not at top of file
mindsdb/__main__.py:22:1: E402 module level import not at top of file
mindsdb/__main__.py:23:1: E402 module level import not at top of file
mindsdb/__main__.py:24:1: E402 module level import not at top of file
mindsdb/__main__.py:25:1: E402 module level import not at top of file
mindsdb/__main__.py:26:1: E402 module level import not at top of file
mindsdb/__main__.py:30:1: E402 module level import not at top of file
mindsdb/__main__.py:31:1: E402 module level import not at top of file
mindsdb/__main__.py:32:1: E402 module level import not at top of file
mindsdb/__main__.py:57:1: C901 'If 57' is too complex (44)
tests/unit/test_executor.py:2:1: F401 'os.path' imported but unused
tests/unit/test_executor.py:8:1: F401 'pytest' imported but unused
tests/unit/test_executor.py:44:17: F541 f-string is missing placeholders
tests/unit/test_executor.py:74:17: F541 f-string is missing placeholders
tests/unit/test_executor.py:144:17: F541 f-string is missing placeholders
tests/unit/test_executor.py:162:17: F541 f-string is missing placeholders
tests/unit/test_executor.py:179:17: F541 f-string is missing placeholders
tests/unit/test_executor.py:209:17: F541 f-string is missing placeholders
tests/unit/test_executor.py:280:17: F541 f-string is missing placeholders
tests/unit/test_executor.py:301:17: F541 f-string is missing placeholders
tests/unit/test_executor.py:315:21: F541 f-string is missing placeholders
tests/unit/test_executor.py:330:21: F541 f-string is missing placeholders
tests/unit/test_executor.py:485:17: F541 f-string is missing placeholders
tests/unit/test_executor.py:521:17: F541 f-string is missing placeholders
tests/unit/test_executor.py:557:17: F541 f-string is missing placeholders
tests/unit/test_executor.py:583:17: F541 f-string is missing placeholders
mindsdb/api/http/namespaces/file.py:29:5: C901 'File.put' is too complex (20)
mindsdb/integrations/handlers/informix_handler/informix_handler.py:4:20: E741 ambiguous variable name 'I'
mindsdb/integrations/handlers/ludwig_handler/test_ludwig.py:12:1: E402 module level import not at top of file
mindsdb/integrations/handlers/ludwig_handler/test_ludwig.py:16:1: E402 module level import not at top of file
mindsdb/integrations/handlers/ludwig_handler/test_ludwig.py:17:1: E402 module level import not at top of file
mindsdb/integrations/handlers/ludwig_handler/test_ludwig.py:18:1: E402 module level import not at top of file
mindsdb/integrations/handlers/ludwig_handler/test_ludwig.py:22:1: E402 module level import not at top of file
mindsdb/integrations/handlers/ludwig_handler/test_ludwig.py:23:1: E402 module level import not at top of file
mindsdb/integrations/handlers/ludwig_handler/test_ludwig.py:24:1: E402 module level import not at top of file
mindsdb/integrations/handlers/ludwig_handler/test_ludwig.py:25:1: E402 module level import not at top of file
mindsdb/integrations/handlers/ludwig_handler/test_ludwig.py:85:9: F841 local variable 'predicted' is assigned to but never used
mindsdb/integrations/handlers/ludwig_handler/test_ludwig.py:94:9: F841 local variable 'predicted' is assigned to but never used
mindsdb/api/mongo/utilities/mongodb_ast.py:4:1: F403 'from mindsdb_sql.parser.ast import *' used; unable to detect undefined names
mindsdb/api/mongo/utilities/mongodb_ast.py:59:21: F405 'OrderBy' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:60:31: F405 'Identifier' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:70:29: F405 'Identifier' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:73:32: F405 'Identifier' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:75:24: F405 'Star' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:84:26: F405 'Identifier' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:86:26: F405 'Identifier' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:88:16: F405 'Select' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:95:26: F405 'Constant' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:98:27: F405 'Constant' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:125:28: F405 'BinaryOperation' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:133:20: F405 'Identifier' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:136:20: F405 'Constant' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:137:23: F405 'BinaryOperation' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:141:30: F405 'BinaryOperation' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:165:25: F405 'Tuple' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:201:24: F405 'BinaryOperation' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:212:20: F405 'BinaryOperation' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:217:24: F405 'Latest' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:221:20: F405 'Constant' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:225:20: F405 'Constant' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:228:20: F405 'Constant' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/utilities/mongodb_ast.py:236:20: F405 'Identifier' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/integrations/handlers/lightwood_handler/setup.py:1:1: F401 'os' imported but unused
mindsdb/integrations/handlers/mlflow_handler/setup.py:3:19: F821 undefined name 'install_deps'
mindsdb/integrations/handlers/mlflow_handler/setup.py:6:10: F821 undefined name 'about'
mindsdb/integrations/handlers/mlflow_handler/setup.py:7:13: F821 undefined name 'about'
mindsdb/integrations/handlers/mlflow_handler/setup.py:8:9: F821 undefined name 'about'
mindsdb/integrations/handlers/mlflow_handler/setup.py:9:18: F821 undefined name 'about'
mindsdb/integrations/handlers/mlflow_handler/setup.py:10:13: F821 undefined name 'about'
mindsdb/integrations/handlers/mlflow_handler/setup.py:11:12: F821 undefined name 'about'
mindsdb/integrations/handlers/mlflow_handler/setup.py:12:18: F821 undefined name 'about'
mindsdb/integrations/handlers/mlflow_handler/setup.py:13:17: F821 undefined name 'about'
mindsdb/integrations/handlers/mlflow_handler/setup.py:14:22: F821 undefined name 'long_description'
mindsdb/api/mongo/responders/list_collections.py:4:1: F403 'from mindsdb_sql.parser.ast import *' used; unable to detect undefined names
mindsdb/api/mongo/responders/list_collections.py:16:21: F405 'Show' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mongo/responders/list_collections.py:16:56: F405 'Identifier' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/api/mysql/mysql_proxy/classes/sql_query.py:16:1: F401 'time' imported but unused
mindsdb/api/mysql/mysql_proxy/classes/sql_query.py:25:1: F401 'mindsdb_sql.parser.ast.CreateTable' imported but unused
mindsdb/api/mysql/mysql_proxy/classes/sql_query.py:66:1: F401 'mindsdb.interfaces.storage.db' imported but unused
mindsdb/api/mysql/mysql_proxy/classes/sql_query.py:304:16: E713 test for membership should be 'not in'
mindsdb/api/mysql/mysql_proxy/classes/sql_query.py:736:5: C901 'SQLQuery.execute_query' is too complex (26)
mindsdb/api/mysql/mysql_proxy/classes/sql_query.py:853:5: C901 'SQLQuery.execute_step' is too complex (158)
mindsdb/api/mysql/mysql_proxy/classes/sql_query.py:1738:5: C901 'SQLQuery.apply_ts_filter' is too complex (48)
mindsdb/api/mysql/mysql_proxy/classes/sql_query.py:1762:32: W605 invalid escape sequence '\d'
mindsdb/api/mysql/mysql_proxy/classes/sql_query.py:1762:40: W605 invalid escape sequence '\d'
mindsdb/api/mysql/mysql_proxy/classes/sql_query.py:1762:48: W605 invalid escape sequence '\d'
mindsdb/api/mysql/mysql_proxy/datahub/datanodes/integration_datanode.py:11:1: F401 'mindsdb.api.mysql.mysql_proxy.datahub.classes.tables_row.TABLES_ROW_TYPE' imported but unused
mindsdb/migrations/versions/2021-11-30_17c3d2384711_init.py:1:1: F401 'sqlalchemy as sa' imported but unused
mindsdb/migrations/versions/2021-11-30_17c3d2384711_init.py:2:1: F401 'alembic.op' imported but unused
mindsdb/migrations/versions/2021-11-30_17c3d2384711_init.py:6:1: F401 'mindsdb.interfaces.storage.db' imported but unused
mindsdb/migrations/versions/2021-11-30_17c3d2384711_init.py:14:1: E402 module level import not at top of file
mindsdb/migrations/versions/2021-11-30_17c3d2384711_init.py:16:1: E402 module level import not at top of file
mindsdb/migrations/versions/2021-11-30_17c3d2384711_init.py:25:1: E402 module level import not at top of file
mindsdb/migrations/versions/2021-11-30_17c3d2384711_init.py:28:1: E402 module level import not at top of file
mindsdb/utilities/cache.py:56:1: F401 'json' imported but unused
mindsdb/utilities/cache.py:62:1: F401 'dill' imported but unused
mindsdb/integrations/handlers/mysql_handler/mysql_handler.py:8:1: F401 'sqlalchemy.create_engine' imported but unused
mindsdb/api/mysql/mysql_proxy/mysql_proxy.py:186:5: C901 'MysqlProxy.handshake' is too complex (19)
mindsdb/api/mysql/mysql_proxy/mysql_proxy.py:633:5: C901 'MysqlProxy.handle' is too complex (21)
mindsdb/integrations/handlers/lightwood_handler/lightwood_handler/functions.py:17:1: F401 'mindsdb.interfaces.model.functions.get_model_record' imported but unused
mindsdb/integrations/handlers/tidb_handler/tests/test_tidb_handler.py:3:1: F401 'pandas as pd' imported but unused
mindsdb/integrations/handlers/tidb_handler/tests/test_tidb_handler.py:51:9: F841 local variable 'result' is assigned to but never used
mindsdb/integrations/libs/base_handler.py:3:1: F401 'pandas as pd' imported but unused
mindsdb/integrations/libs/base_handler.py:4:1: F401 'mindsdb_sql.parser.ast.Join' imported but unused
mindsdb/integrations/handlers/airtable_handler/airtable_handler.py:73:13: F841 local variable 'e' is assigned to but never used
mindsdb/integrations/handlers/supabase_handler/tests/test_supabase_handler.py:51:9: F841 local variable 'result' is assigned to but never used
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_parser.py:34:5: C901 'MongodbParser.process' is too complex (24)
tests/unit/test_cache.py:10:1: F401 'mindsdb.utilities.cache.get_cache' imported but unused
mindsdb/integrations/handlers/d0lt_handler/tests/test_d0lt_handler.py:23:5: F811 redefinition of unused 'test_0_connect' from line 20
mindsdb/integrations/handlers/elasticsearch_handler/elasticsearch_handler.py:146:17: F841 local variable 'e' is assigned to but never used
mindsdb/integrations/handlers/trino_handler/trino_handler.py:7:1: F401 'pyhive.trino' imported but unused
mindsdb/integrations/handlers/trino_handler/trino_handler.py:8:1: F401 'trino.auth.KerberosAuthentication' imported but unused
mindsdb/integrations/handlers/trino_handler/trino_handler.py:17:1: F401 '.trino_config_provider.TrinoConfigProvider' imported but unused
mindsdb/integrations/handlers/trino_handler/trino_handler.py:164:35: W605 invalid escape sequence '\('
mindsdb/integrations/handlers/trino_handler/trino_handler.py:164:39: W605 invalid escape sequence '\)'
mindsdb/integrations/handlers/mysql_handler/tests/test_mysql_handler.py:171:57: F821 undefined name 'df'
mindsdb/integrations/handlers/mysql_handler/tests/test_mysql_handler.py:174:63: F821 undefined name 'df'
mindsdb/integrations/handlers/mlflow_handler/tests/test_mlflow_handler.py:44:9: E722 do not use bare 'except'
mindsdb/integrations/handlers/mlflow_handler/tests/test_mlflow_handler.py:64:9: F841 local variable 'predicted' is assigned to but never used
mindsdb/integrations/handlers/mlflow_handler/tests/test_mlflow_handler.py:72:9: F841 local variable 'predicted' is assigned to but never used
mindsdb/integrations/handlers/mlflow_handler/tests/test_mlflow_handler.py:82:9: E722 do not use bare 'except'
mindsdb/migrations/migrate.py:3:1: F401 'alembic.command.autogen' imported but unused
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_render.py:3:1: F403 'from mindsdb_sql.parser.ast import *' used; unable to detect undefined names
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_render.py:10:29: F405 'Select' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_render.py:14:5: C901 'MongodbRender.select' is too complex (23)
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_render.py:16:44: F405 'Identifier' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_render.py:35:36: F405 'Star' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_render.py:39:36: F405 'Identifier' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_render.py:53:38: F405 'Constant' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_render.py:108:31: F405 'BinaryOperation' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_render.py:139:29: F405 'Identifier' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_render.py:142:33: F405 'Constant' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_render.py:157:35: F405 'Tuple' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_render.py:184:29: F405 'Identifier' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_render.py:186:31: F405 'Latest' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_render.py:187:20: F541 f-string is missing placeholders
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_render.py:188:31: F405 'Constant' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/integrations/handlers/mongodb_handler/utils/mongodb_render.py:190:31: F405 'TypeCast' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/integrations/handlers/mariadb_handler/tests/test_mariadb_handler.py:3:1: F401 'pandas as pd' imported but unused
mindsdb/integrations/handlers/mariadb_handler/tests/test_mariadb_handler.py:51:9: F841 local variable 'result' is assigned to but never used
mindsdb/api/mysql/mysql_proxy/executor/executor_commands.py:49:1: F401 'mindsdb.api.mysql.mysql_proxy.libs.constants.mysql.ERR' imported but unused
mindsdb/api/mysql/mysql_proxy/executor/executor_commands.py:56:1: F401 'mindsdb.api.mysql.mysql_proxy.utilities.ErDbDropDelete' imported but unused
mindsdb/api/mysql/mysql_proxy/executor/executor_commands.py:70:1: F401 'mindsdb.api.mysql.mysql_proxy.utilities.functions.get_column_in_case' imported but unused
mindsdb/api/mysql/mysql_proxy/executor/executor_commands.py:80:1: F401 'mindsdb.interfaces.model.functions.get_model_record' imported but unused
mindsdb/api/mysql/mysql_proxy/executor/executor_commands.py:91:5: C901 'ExecuteCommands.execute_command' is too complex (83)
tests/unit/test_mongodb_server.py:4:1: F401 'unittest' imported but unused
tests/unit/test_mongodb_server.py:169:9: F841 local variable 'res' is assigned to but never used
mindsdb/integrations/handlers/monetdb_handler/monetdb_handler.py:18:1: F403 'from .utils.monet_get_id import *' used; unable to detect undefined names
mindsdb/integrations/handlers/monetdb_handler/monetdb_handler.py:166:18: F405 'schema_id' may be undefined, or defined from star imports: .utils.monet_get_id
mindsdb/integrations/handlers/monetdb_handler/monetdb_handler.py:190:17: F405 'table_id' may be undefined, or defined from star imports: .utils.monet_get_id
mindsdb/migrations/env.py:20:1: E402 module level import not at top of file
mindsdb/migrations/env.py:23:1: F401 'mindsdb.__init__' imported but unused
mindsdb/migrations/env.py:23:1: E402 module level import not at top of file
mindsdb/migrations/env.py:24:1: E402 module level import not at top of file
mindsdb/api/mongo/responders/insert.py:8:1: F401 'mindsdb.api.mysql.mysql_proxy.libs.constants.response_type.RESPONSE_TYPE' imported but unused
mindsdb/api/mongo/responders/insert.py:27:5: C901 'Responce._result' is too complex (26)
mindsdb/integrations/handlers/pinot_handler/pinot_handler.py:177:9: F841 local variable 'e' is assigned to but never used
mindsdb/integrations/handlers/pinot_handler/pinot_handler.py:202:9: F841 local variable 'e' is assigned to but never used
mindsdb/api/mongo/responders/list_databases.py:1:1: F403 'from mindsdb_sql.parser.ast import *' used; unable to detect undefined names
mindsdb/api/mongo/responders/list_databases.py:13:21: F405 'Show' may be undefined, or defined from star imports: mindsdb_sql.parser.ast
mindsdb/migrations/versions/2022-07-22_6e834843e7e9_training_time.py:11:1: F401 'mindsdb.interfaces.storage.db' imported but unused
mindsdb/api/mysql/mysql_proxy/datahub/datanodes/mindsdb_datanode.py:5:1: F401 'mindsdb_sql.parse_sql' imported but unused
mindsdb/migrations/versions/2022-07-08_999bceb904df_integration_args.py:14:1: F401 'mindsdb.interfaces.storage.db' imported but unused
mindsdb/integrations/handlers/monetdb_handler/utils/monet_get_id.py:21:5: E722 do not use bare 'except'
mindsdb/integrations/handlers/monetdb_handler/utils/monet_get_id.py:46:5: E722 do not use bare 'except'
mindsdb/api/mysql/mysql_proxy/data_types/mysql_packets/binary_resultset_row_package.py:18:1: F401 'mindsdb.api.mysql.mysql_proxy.libs.constants.mysql.NULL_VALUE' imported but unused
mindsdb/api/mysql/mysql_proxy/data_types/mysql_packets/binary_resultset_row_package.py:28:5: C901 'BinaryResultsetRowPacket.setup' is too complex (20)
mindsdb/migrations/versions/2022-01-26_47f97b83cee4_views.py:11:1: F401 'mindsdb.interfaces.storage.db' imported but unused
tests/unit/executor_test_base.py:163:13: F841 local variable 'dict_arr' is assigned to but never used
tests/unit/executor_test_base.py:295:13: E722 do not use bare 'except'
mindsdb/api/http/start.py:33:1: C901 'start' is too complex (25)
mindsdb/api/http/namespaces/predictor.py:10:1: F401 'mindsdb.api.http.namespaces.entitites.predictor_metadata.put_predictor_params' imported but unused
mindsdb/api/http/namespaces/predictor.py:17:1: F401 'mindsdb.utilities.log.log' imported but unused
mindsdb/integrations/handlers/couchbase_handler/couchbase_handler.py:3:1: F401 'typing.Optional' imported but unused
mindsdb/integrations/handlers/couchbase_handler/couchbase_handler.py:10:1: F401 'couchbase.exceptions.QueryErrorContext' imported but unused
mindsdb/integrations/handlers/couchbase_handler/couchbase_handler.py:16:1: F401 'couchbase.n1ql.N1QLQuery' imported but unused
mindsdb/integrations/handlers/couchbase_handler/couchbase_handler.py:19:1: F401 'couchbase.options.QueryOptions' imported but unused
mindsdb/integrations/handlers/couchbase_handler/couchbase_handler.py:60:9: F841 local variable 'bucket_name' is assigned to but never used
mindsdb/integrations/handlers/couchbase_handler/couchbase_handler.py:62:9: F841 local variable 'timeout_opts' is assigned to but never used
mindsdb/api/mysql/mysql_proxy/data_types/mysql_packets/switch_auth_packet.py:23:9: F841 local variable 'status' is assigned to but never used
mindsdb/api/mysql/mysql_proxy/external_libs/mysql_scramble.py:40:18: F821 undefined name 'xrange'
mindsdb/api/mysql/mysql_proxy/external_libs/mysql_scramble.py:41:17: F821 undefined name 'unicode'
mindsdb/api/mysql/mysql_proxy/external_libs/mysql_scramble.py:42:17: F821 undefined name 'long'
mindsdb/api/mysql/mysql_proxy/external_libs/mysql_scramble.py:43:16: F821 undefined name 'basestring'
mindsdb/integrations/handlers/ludwig_handler/setup.py:3:19: F821 undefined name 'install_deps'
mindsdb/integrations/handlers/ludwig_handler/setup.py:6:10: F821 undefined name 'about'
mindsdb/integrations/handlers/ludwig_handler/setup.py:7:13: F821 undefined name 'about'
mindsdb/integrations/handlers/ludwig_handler/setup.py:8:9: F821 undefined name 'about'
mindsdb/integrations/handlers/ludwig_handler/setup.py:9:18: F821 undefined name 'about'
mindsdb/integrations/handlers/ludwig_handler/setup.py:10:13: F821 undefined name 'about'
mindsdb/integrations/handlers/ludwig_handler/setup.py:11:12: F821 undefined name 'about'
mindsdb/integrations/handlers/ludwig_handler/setup.py:12:18: F821 undefined name 'about'
mindsdb/integrations/handlers/ludwig_handler/setup.py:13:17: F821 undefined name 'about'
mindsdb/integrations/handlers/ludwig_handler/setup.py:14:22: F821 undefined name 'long_description'

```

--- a/.flake8
+++ b/.flake8
@@ -1,6 +1,75 @@
 [flake8]
 max-line-length = 88
 select = C,E,F,W,B,B9
-ignore = E203, E501, W503
-exclude = __init__.py
+ignore = E203, E402, E501, E713, W503, C901, F401, F403, F405, F541, F821,
+exclude =
+  __init__.py,
+  # TODO: clean up this code
+
+  mindsdb/api/mongo/*
+  mindsdb/api/mysql/mysql_proxy/*
+  mindsdb/api/mysql/mysql_proxy/external_libs/mysql_scramble.py
+  mindsdb/integrations/handlers/access_handler/access_handler.py
+  mindsdb/integrations/handlers/airtable_handler/airtable_handler.py
+  mindsdb/integrations/handlers/bigquery_handler/tests/test_bigquery_handler.py
+  mindsdb/integrations/handlers/couchbase_handler/couchbase_handler.py
+  mindsdb/integrations/handlers/couchbase_handler/tests/test_couchbase_handler.py
+  mindsdb/integrations/handlers/crate_handler/crate_handler.py
+  mindsdb/integrations/handlers/crate_handler/tests/test_crate_handler.py
+  mindsdb/integrations/handlers/d0lt_handler/tests/test_d0lt_handler.py
+  mindsdb/integrations/handlers/databricks_handler/databricks_handler.py
+  mindsdb/integrations/handlers/datastax_handler/datastax_handler.py
+  mindsdb/integrations/handlers/db2_handler/*
+  mindsdb/integrations/handlers/druid_handler/druid_handler.py
+  mindsdb/integrations/handlers/dynamodb_handler/dynamodb_handler.py
+  mindsdb/integrations/handlers/elasticsearch_handler/elasticsearch_handler.py
+  mindsdb/integrations/handlers/firebird_handler
+  mindsdb/integrations/handlers/firebird_handler/firebird_handler.py
+  mindsdb/integrations/handlers/hive_handler/tests/test_hive_handler.py
+  mindsdb/integrations/handlers/informix_handler/informix_handler.py
+  mindsdb/integrations/handlers/informix_handler/tests/test_informix_handler.py
+  mindsdb/integrations/handlers/lightwood_handler/*
+  mindsdb/integrations/handlers/ludwig_handler/test_ludwig.py
+  mindsdb/integrations/handlers/mariadb_handler/tests/test_mariadb_handler.py
+  mindsdb/integrations/handlers/matrixone_handler/*
+  mindsdb/integrations/handlers/mlflow_handler/*
+  mindsdb/integrations/handlers/monetdb_handler/*
+  mindsdb/integrations/handlers/mongodb_handler/*
+  mindsdb/integrations/handlers/mssql_handler/*
+  mindsdb/integrations/handlers/mysql_handler/tests/test_mysql_handler.py
+  mindsdb/integrations/handlers/oracle_handler/oracle_handler.py
+  mindsdb/integrations/handlers/pinot_handler/pinot_handler.py
+  mindsdb/integrations/handlers/postgres_handler/tests/test_postgres_handler.py
+  mindsdb/integrations/handlers/questdb_handler/*
+  mindsdb/integrations/handlers/redshift_handler/*
+  mindsdb/integrations/handlers/s3_handler/s3_handler.py
+  mindsdb/integrations/handlers/singlestore_handler/*
+  mindsdb/integrations/handlers/snowflake_handler/snowflake_handler.py
+  mindsdb/integrations/handlers/sqlite_handler/sqlite_handler.py
+  mindsdb/integrations/handlers/supabase_handler/tests/test_supabase_handler.py
+  mindsdb/integrations/handlers/tidb_handler/*
+  mindsdb/integrations/handlers/timescaledb_handler/*
+  mindsdb/integrations/handlers/trino_handler/trino_handler.py
+  mindsdb/integrations/handlers/vertica_handler/*
+  mindsdb/integrations/handlers/view_handler/view_handler.py
+  mindsdb/integrations/handlers/yugabyte_handler/*
+  mindsdb/integrations/libs/response.py
+  mindsdb/integrations/utilities/utils.py
+  mindsdb/interfaces/model/model_controller.py
+  mindsdb/migrations/env.py
+  mindsdb/migrations/versions/2021-11-30_17c3d2384711_init.py
+  mindsdb/migrations/versions/2022-01-26_47f97b83cee4_views.py
+  mindsdb/migrations/versions/2022-08-25_6a54ba55872e_view_integration.py
+  mindsdb/utilities/cache.py
+  mindsdb/utilities/log.py
+  mindsdb/utilities/loop.py
+  tests/handler_tests/test_storage_handler.py
+  tests/integration_tests/flows/test_mysql_bin_api.py
+  tests/prediction_latency_test/*
+  tests/unit/executor_test_base.py
+  tests/unit/test_cache.py
+  tests/unit/test_executor.py
+  tests/unit/test_mongodb_handler.py
+  tests/unit/test_mongodb_server.py
+
 max-complexity = 18

--- a/.flake8
+++ b/.flake8
@@ -1,6 +1,6 @@
 [flake8]
-# E701: multiple statements on one line (colon)
-ignore = E701
-max-line-length = 160
-# W503: Line break occurred before a binary operator
-extend-ignore = W503
+max-line-length = 88
+select = C,E,F,W,B,B9
+ignore = E203, E501, W503
+exclude = __init__.py
+max-complexity = 18

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -1,0 +1,27 @@
+exclude: '^databricks_src|^data_engineering/'
+repos:
+  - repo: https://github.com/ambv/black
+    rev: 22.3.0
+    hooks:
+      - id: black
+  - repo: https://github.com/pycqa/flake8
+    rev: 4.0.1
+    hooks:
+      - id: flake8
+        additional_dependencies: [ flake8-typing-imports==1.10.0 ]
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.3.0
+    hooks:
+      - id: check-added-large-files
+      - id: check-toml
+      - id: check-yaml
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: check-merge-conflict
+      - id: check-yaml
+        args: [ --unsafe ]
+  - repo: https://github.com/pre-commit/mirrors-isort
+    rev: v5.10.1
+    hooks:
+      - id: isort
+        args: [ "--filter-files", "--profile", "black" ]

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -15,11 +15,11 @@ repos:
       - id: check-added-large-files
       - id: check-toml
       - id: check-yaml
+        exclude: "^helm/"
+        args: [--unsafe]
       - id: trailing-whitespace
       - id: end-of-file-fixer
       - id: check-merge-conflict
-      - id: check-yaml
-        args: [ --unsafe ]
   - repo: https://github.com/pre-commit/mirrors-isort
     rev: v5.10.1
     hooks:

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -1,0 +1,6 @@
+black==22.3.0
+flake8==5.0.4
+invoke>=1.7.1
+isort==5.10.1
+mypy>=0.971
+pre-commit>=2.16.0

